screens/ProductionScreens/BudgetAddExpense.jsx

```diff
@@ -0,0 +1,176 @@
+                    );
+                }
+            }
+        });
+    };
+
+    const setReceiptCamera = async () => {
+        const cameraStatus = await requestCameraPermissionsAsync();
+        if (cameraStatus.status != "granted") {
+            console.log("Camera Permission Denied");
+            return;
+        }
+        const result = await launchCameraAsync({ quality: 0.1 });
+
+        if (!result.canceled) {
+            try {
+                const response = await fetch(result.assets[0].uri);
+                const blob = await response.blob();
+                const extension = result.assets[0].uri.substring(
+                    result.assets[0].uri.lastIndexOf(".")
+                );
+                const snapshot = await uploadBytes(
+                    ref(
+                        storageRef,
+                        "receipts/" +
+                            expenseID +
+                            extension
+                    ),
+                    blob
+                );
+                const url = await getDownloadURL(snapshot.ref);
+                setReceiptURL(url);
+            } catch (error) {
+                setAlert(
+                    "Error uploading receipt",
+                    "bg-red-500",
+                    icon({ name: "circle-exclamation" })
+                );
+            }
+        }
+    };
+
+    return (
+        <View className="py-2 flex justify-center items-center">
+            <Text className="self-center text-3xl font-extrabold">Add Expense</Text>
+            <ScrollView className="h-5/6 w-10/12">
+                <View>
+                    {/* TODO: Select Budget */}
+                    <Text className="text-lg font-semibold text-center">Select Budget</Text>
+                </View>
+                <View>
+                    <Text className="text-lg font-semibold text-center">Reference</Text>
+                    <FormField
+                        value={reference}
+                        placeholder="Reference"
+                        onChangeText={setReference}
+                    />
+                </View>
+                <View>
+                    <Text className="text-lg font-semibold text-center">Description</Text>
+                    <FormField
+                        value={description}
+                        placeholder="Description"
+                        onChangeText={setDescription}
+                        multiline
+                        extraClassName="h-20"
+                    />
+                </View>
+                <View>
+                    <Text className="text-lg font-semibold text-center">Cost</Text>
+                    <FormField value={cost} placeholder="Cost (£)" onChangeText={setCost} />
+                </View>
+                <View className="h-80">
+                    <Text className="text-lg font-semibold text-center">Receipt</Text>
+                    <ReceiptViewer recieptURL={receiptURL} />
```

Comment:
Kinda feel like I find myself wanting to tap on the empty receipt box to add a pic? but I realise that would be a long fix for a fairly particular issue

Reply:
Yeah see I'd like to, but I can't find anything saying about the multiple options (camera/library) popup

Reply:
Only thing I've seen on this front is this [action sheet thing](https://github.com/expo/react-native-action-sheet). Don't think it would need to be this PR tho if its even any good

Reply:
Yeah that does look useful tbf, defo something to keep in mind! Maybe later we add that in?